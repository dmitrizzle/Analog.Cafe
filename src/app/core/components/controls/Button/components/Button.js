import { Button } from "@roast-cms/react-button-beans"
import Loadable from "react-loadable"
import React from "react"

const Loader = Loadable({
  loader: () => import("../../../icons/Loader"),
  loading: () => null,
  delay: 100
})

export default props => (
  <Button loaderComponent={props.loading ? Loader : null} {...props}>
    {props.children}
  </Button>
)
