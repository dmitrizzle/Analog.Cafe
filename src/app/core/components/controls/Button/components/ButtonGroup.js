import styled from "styled-components"

export default styled.div`
  text-align: center;
  padding: ${props => props.theme.size.block.padding}em 0
    ${props => props.theme.size.block.spacing * 4}em;
`
