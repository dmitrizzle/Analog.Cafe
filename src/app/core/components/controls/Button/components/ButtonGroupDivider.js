import styled from "styled-components"

export default styled.div`
  border-bottom: ${props => props.theme.elements.thickBorder};
  width: 100%;
`
