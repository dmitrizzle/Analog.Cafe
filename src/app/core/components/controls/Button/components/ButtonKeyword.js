import styled from "styled-components"

export default styled.span`
  color: ${props => {
    if (props.branded) return props.theme.color.foreground()
    return props.theme.color.brand()
  }};
`
