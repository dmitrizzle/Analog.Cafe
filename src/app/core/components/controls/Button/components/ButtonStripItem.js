import { Item } from "@roast-cms/react-button-beans"
import React from "react"

import Link from "../../Link"

export default props => (
  <Item linkComponent={Link} {...props}>
    {props.children}
  </Item>
)
