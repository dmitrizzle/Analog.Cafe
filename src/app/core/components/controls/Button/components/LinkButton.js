import { Button, LinkButton } from "@roast-cms/react-button-beans"
import React from "react"

import Link from "../../Link"

const CommonLink = props => (
  <LinkButton linkComponent={Link} {...props}>
    {props.children}
  </LinkButton>
)
const EmailLink = props => (
  <a href={props.to} style={{ textDecoration: "none" }}>
    <Button {...props} />
  </a>
)

export default props => {
  return props.to.includes("mailto:") ? (
    <EmailLink {...props} />
  ) : (
    <CommonLink {...props} />
  )
}
