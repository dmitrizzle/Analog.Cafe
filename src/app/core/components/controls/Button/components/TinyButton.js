import { TinyButton } from "@roast-cms/react-button-beans"
import React from "react"

import Link from "../../Link"

export default props => (
  <TinyButton linkComponent={Link} {...props}>
    {props.children}
  </TinyButton>
)
