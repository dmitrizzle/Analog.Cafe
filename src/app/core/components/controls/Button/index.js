import {
  Button as ButtonInit,
  TinyButton as TinyButtonInit,
  LinkButton as LinkButtonInit
} from "@roast-cms/react-button-beans"
import Loadable from "react-loadable"
import React from "react"

import Link from "../Link"

// direct exports
export { ButtonStyles } from "@roast-cms/react-button-beans"

const Loader = Loadable({
  loader: () => import("../../icons/Loader"),
  loading: () => null,
  delay: 100
})

export const Button = props => (
  <ButtonInit loaderComponent={props.loading ? Loader : null} {...props}>
    {props.children}
  </ButtonInit>
)
export const TinyButton = props => (
  <TinyButtonInit linkComponent={Link} {...props}>
    {props.children}
  </TinyButtonInit>
)
export const LinkButton = props => (
  <LinkButtonInit linkComponent={Link} {...props}>
    {props.children}
  </LinkButtonInit>
)
