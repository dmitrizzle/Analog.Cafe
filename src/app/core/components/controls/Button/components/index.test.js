import "jest-styled-components"

import { ThemeProvider } from "styled-components"
import { shallow } from "enzyme"
import React from "react"

import { APP_THEME } from "../../../../../../constants"
import {
  FacebookLinkButton,
  InstagramLinkButton,
  TwitterLinkButton,
  FeedlyLinkButton
} from "./SocialButtons"
import Button from "./Button"
import ButtonGroup from "./ButtonGroup"
import ButtonGroupDivider from "./ButtonGroupDivider"
import ButtonKeyword from "./ButtonKeyword"
import ButtonStripItem from "./ButtonStripItem"
import LinkButton from "./LinkButton"
import TinyButton from "./TinyButton"

it("Render Button, matches snapshot", () => {
  const element = shallow(<Button />)
  expect(element).toMatchSnapshot()
})

it("Render ButtonGroup, matches snapshot", () => {
  const element = shallow(
    <ThemeProvider theme={APP_THEME}>
      <ButtonGroup />
    </ThemeProvider>
  )
  expect(element).toMatchSnapshot()
})

it("Render ButtonGroupDivider, matches snapshot", () => {
  const element = shallow(
    <ThemeProvider theme={APP_THEME}>
      <ButtonGroupDivider />
    </ThemeProvider>
  )
  expect(element).toMatchSnapshot()
})

it("Render ButtonKeyword, matches snapshot", () => {
  const element = shallow(
    <ThemeProvider theme={APP_THEME}>
      <ButtonKeyword />
    </ThemeProvider>
  )
  expect(element).toMatchSnapshot()
})

it("Render ButtonStripItem, matches snapshot", () => {
  const element = shallow(<ButtonStripItem />)
  expect(element).toMatchSnapshot()
})

it("Render LinkButton, matches snapshot", () => {
  const element = shallow(<LinkButton />)
  expect(element).toMatchSnapshot()
})

it("Render FeedlyLinkButton, matches snapshot", () => {
  const element = shallow(<FeedlyLinkButton />)
  expect(element).toMatchSnapshot()
})
it("Render TwitterLinkButton, matches snapshot", () => {
  const element = shallow(<TwitterLinkButton />)
  expect(element).toMatchSnapshot()
})
it("Render FacebookLinkButton, matches snapshot", () => {
  const element = shallow(<FacebookLinkButton />)
  expect(element).toMatchSnapshot()
})
it("Render InstagramLinkButton, matches snapshot", () => {
  const element = shallow(<InstagramLinkButton />)
  expect(element).toMatchSnapshot()
})

it("Render TinyButton, matches snapshot", () => {
  const element = shallow(<TinyButton />)
  expect(element).toMatchSnapshot()
})
