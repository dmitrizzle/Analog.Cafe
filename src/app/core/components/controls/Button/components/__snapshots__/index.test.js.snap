// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ButtonGroup, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
        Object {
            "color": Object {
              "background": [Function],
              "brand": [Function],
              "foreground": [Function],
              "highlight": [Function],
            },
            "effects": Object {
              "borderRadius": Object {
                "med": 0.5,
                "small": [Function],
              },
            },
            "elements": Object {
              "thickBorder": [Function],
            },
            "layer": Object {
              "card": 30,
              "nav": 20,
              "overlay": 40,
              "tuck": -1,
              "up": 10,
            },
            "opacity": Object {
              "half": 0.5,
              "least": 0.125,
              "most": 0.85,
            },
            "size": Object {
              "block": Object {
                "border": 8,
                "column": Object {
                  "l": 820,
                  "m": 750,
                },
                "padding": 1.5,
                "spacing": 1,
              },
              "breakpoint": Object {
                "exact": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "max": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "min": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "stops": Object {
                  "max": 1600,
                  "min": 320,
                },
              },
              "font": Object {
                "auto": Array [
                  "
                ",
                  "
        		@media (min-width: ",
                  "32.5625",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "32.5",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "20",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (min-width: ",
                  "100.0625",
                  "em) {
        			",
                  "font-size:	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  ";
              ",
                ],
                "l": 23,
                "m": 20,
                "make": Object {
                  "larger": 3,
                  "normal": 1,
                  "smaller": 0.85,
                  "tiny": 0.65,
                },
                "s": 18,
                "xs": 17,
              },
            },
            "typography": Object {
              "text": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Lora', Georgia, serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                ",
                ],
                "font": "'Lora', Georgia, serif",
                "lineHeight": 1.75,
              },
              "title": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Exo 2', Arial, sans-serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                  font-weight: ",
                  "600",
                  ";
                  margin: 0;
                ",
                ],
                "font": "'Exo 2', Arial, sans-serif",
                "lineHeight": 1.15,
              },
            },
          }
    }
>
    <styled.div />
</ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render ButtonGroupDivider, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
        Object {
            "color": Object {
              "background": [Function],
              "brand": [Function],
              "foreground": [Function],
              "highlight": [Function],
            },
            "effects": Object {
              "borderRadius": Object {
                "med": 0.5,
                "small": [Function],
              },
            },
            "elements": Object {
              "thickBorder": [Function],
            },
            "layer": Object {
              "card": 30,
              "nav": 20,
              "overlay": 40,
              "tuck": -1,
              "up": 10,
            },
            "opacity": Object {
              "half": 0.5,
              "least": 0.125,
              "most": 0.85,
            },
            "size": Object {
              "block": Object {
                "border": 8,
                "column": Object {
                  "l": 820,
                  "m": 750,
                },
                "padding": 1.5,
                "spacing": 1,
              },
              "breakpoint": Object {
                "exact": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "max": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "min": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "stops": Object {
                  "max": 1600,
                  "min": 320,
                },
              },
              "font": Object {
                "auto": Array [
                  "
                ",
                  "
        		@media (min-width: ",
                  "32.5625",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "32.5",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "20",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (min-width: ",
                  "100.0625",
                  "em) {
        			",
                  "font-size:	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  ";
              ",
                ],
                "l": 23,
                "m": 20,
                "make": Object {
                  "larger": 3,
                  "normal": 1,
                  "smaller": 0.85,
                  "tiny": 0.65,
                },
                "s": 18,
                "xs": 17,
              },
            },
            "typography": Object {
              "text": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Lora', Georgia, serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                ",
                ],
                "font": "'Lora', Georgia, serif",
                "lineHeight": 1.75,
              },
              "title": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Exo 2', Arial, sans-serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                  font-weight: ",
                  "600",
                  ";
                  margin: 0;
                ",
                ],
                "font": "'Exo 2', Arial, sans-serif",
                "lineHeight": 1.15,
              },
            },
          }
    }
>
    <styled.div />
</ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render ButtonKeyword, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
        Object {
            "color": Object {
              "background": [Function],
              "brand": [Function],
              "foreground": [Function],
              "highlight": [Function],
            },
            "effects": Object {
              "borderRadius": Object {
                "med": 0.5,
                "small": [Function],
              },
            },
            "elements": Object {
              "thickBorder": [Function],
            },
            "layer": Object {
              "card": 30,
              "nav": 20,
              "overlay": 40,
              "tuck": -1,
              "up": 10,
            },
            "opacity": Object {
              "half": 0.5,
              "least": 0.125,
              "most": 0.85,
            },
            "size": Object {
              "block": Object {
                "border": 8,
                "column": Object {
                  "l": 820,
                  "m": 750,
                },
                "padding": 1.5,
                "spacing": 1,
              },
              "breakpoint": Object {
                "exact": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "max": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "min": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "stops": Object {
                  "max": 1600,
                  "min": 320,
                },
              },
              "font": Object {
                "auto": Array [
                  "
                ",
                  "
        		@media (min-width: ",
                  "32.5625",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "32.5",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "20",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (min-width: ",
                  "100.0625",
                  "em) {
        			",
                  "font-size:	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  ";
              ",
                ],
                "l": 23,
                "m": 20,
                "make": Object {
                  "larger": 3,
                  "normal": 1,
                  "smaller": 0.85,
                  "tiny": 0.65,
                },
                "s": 18,
                "xs": 17,
              },
            },
            "typography": Object {
              "text": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Lora', Georgia, serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                ",
                ],
                "font": "'Lora', Georgia, serif",
                "lineHeight": 1.75,
              },
              "title": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Exo 2', Arial, sans-serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                  font-weight: ",
                  "600",
                  ";
                  margin: 0;
                ",
                ],
                "font": "'Exo 2', Arial, sans-serif",
                "lineHeight": 1.15,
              },
            },
          }
    }
>
    <styled.span />
</ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render ButtonStripItem, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": undefined,
      "linkComponent": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "linkComponent": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render FacebookLinkButton, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown />,
        " ",
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " ",
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown />,
          " ",
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render FeedlyLinkButton, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown />,
        " ",
        undefined,
      ],
      "inverse": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " ",
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown />,
          " ",
          undefined,
        ],
        "inverse": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render InstagramLinkButton, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown />,
        "  ",
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "  ",
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown />,
          "  ",
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "  ",
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render LinkButton, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": undefined,
      "linkComponent": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "linkComponent": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render TinyButton, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "linkComponent": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "linkComponent": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render TwitterLinkButton, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown />,
        " ",
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " ",
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown />,
          " ",
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
